frontend/pkg/frontend/frontend.go

```diff
@@ -168,6 +166,14 @@ func (f *Frontend) Join() {
 }
 
 func (f *Frontend) CheckReady() bool {
+	// Verify the DB is available and accessible
+	result, err := f.dbClient.DBConnectionTest(context.Background())
```

Comment:
Should we put a context timeout here, or will the overarching liveness/readiness probes cancel the http context should they exceed the threshold?

Reply:
A timeout makes sense, I'm not sure if they get canceled, so I'll try reading up

Reply:
I suggest have the function take a `context.Context` argument and decide the timeout in the endpoint handler.  For now we could just plumb the request's context into here.